### PR TITLE
[HW-2] Implement my own array methods

### PR DESCRIPTION
**Pull request by DashaSledopit**

### CODE REVIEW

File: HW02/Nastja_Kireeva/hw02.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      end
+      new_array
+    end
+
+    def my_empty?
+      if length.zero?
+        true
+      else
+        false
+      end
```
Comment (B0RlS) [inline HW02/Nastja_Kireeva/hw02.rb:37]: Use ternary conditional operation

File: HW02/Nastja_Kireeva/hw02.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
+module ArrayModule
+  refine Array do
+    def my_each
+      counter = 0
+      while counter < length
+        yield(self[counter])
+        counter += 1
+      end
+      self
```
Comment (B0RlS) [inline HW02/Nastja_Kireeva/hw02.rb:9]: How do you think you can do this in a different way, using a different loop? You can write your solution in the comment here

Comment (DashaSledopit) [reply HW02/Nastja_Kireeva/hw02.rb:9]: for counter in (0...length)

File: HW02/Nastja_Kireeva/hw02.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+module ArrayModule
```
Comment (B0RlS) [inline HW02/Nastja_Kireeva/hw02.rb:1]: Could you change the file name please?

Comment (DashaSledopit) [reply HW02/Nastja_Kireeva/hw02.rb:1]: File or module name? I don’t understand. And if file name, what’s wrong with it?

Comment (B0RlS) [reply HW02/Nastja_Kireeva/hw02.rb:1]: You should give the filename the same as the module name.

File: HW02/Nastja_Kireeva/array_module.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+using ArrayModule
+
+array = [1, 2, 3, 4, 5, 6]
+
+array.my_each { |element| p element }
+p array.my_map { |element| element * 3 }
+p array.my_select(&:odd?)
+p array.my_reject(&:odd?)
+p array.my_empty?
+p array.my_count { |element| (element % 3).zero? }
```
Comment (B0RlS) [inline HW02/Nastja_Kireeva/array_module.rb:67]: This is not a good practice, it is better to take it out of the file. 
Could you move this to another file please? (For example create runner.rb)

**Comment by B0RlS** [comment]
Everything else LGTM

**Review by vladhilko — APPROVED** [review]

**Review by B0RlS — APPROVED** [review]

**Pull request merged**